### PR TITLE
[RFC] dts: Add a pykwalify schema file for dts bindings

### PR DESCRIPTION
**Pull request by galak**
Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

### CODE REVIEW

**Comment by galak** [comment]
Posting this to generate some discussion on how to handle this.

Do we merge this with dts/binding-template.yaml?  Ok, if dts/binding-template.yaml moves under dts/scripts?

**Comment by erwango** [comment]
> Do we merge this with dts/binding-template.yaml? 

Would make sense

> Ok, if dts/binding-template.yaml moves under dts/scripts?

I'd say no, as I doubt anyone will see it there.

**Comment by galak** [comment]
> > Ok, if dts/binding-template.yaml moves under dts/scripts?
> 
> I'd say no, as I doubt anyone will see it there.

The issue I have is wanting to keep "edtlib" contained together.  So if/when we start having the python code call pykwalify to do initial schema validation having it pull that from dts/binding-template.yaml doesn't make sense to me.    Will be interested in @mbolivar-nordic take from how west deals with this.

**Comment by mbolivar-nordic** [comment]
> Will be interested in @mbolivar-nordic take from how west deals with this.

West has a couple of schema files in the source distribution:

https://github.com/zephyrproject-rtos/west/blob/master/src/west/manifest-schema.yml
https://github.com/zephyrproject-rtos/west/blob/master/src/west/west-commands-schema.yml

West also has RST documentation for the manifest format:

https://github.com/zephyrproject-rtos/zephyr/blob/master/doc/guides/west/manifest.rst

It mentions the pykwalify schema but doesn't rely on it for documentation. The format is explained in standalone docs.

Right now, for edtlib, binding-template.yaml is the bindings file format documentation. I think it makes sense to keep it a separate file for now.

If we ever extract edtlib into a separate package we put on PyPI, we can link to its documentation with a hyperlink from the Zephyr docs. Users can find information about formats there. For now, I think keeping the same west approach where documentation is in zephyr/doc/ (or included into it, like we do with binding-template.yaml) makes sense.

File: scripts/dts/binding-schema.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    bus:
+      type: str
+
+    on-bus:
+      type: str
+
+    properties:
+      type: map
+      mapping:
+        regex;([a-zA-Z0-9,._+*#?@-]+):
```
Comment (mbolivar-nordic) [inline scripts/dts/binding-schema.yaml:22]: Cool. I didn't know pykwalify could do that.

**Review by mbolivar-nordic — COMMENTED** [review]
@galak the schema itself looks nice.

**Pull request closed without merging**